src/racer/nameres.rs

```diff
@@ -1819,6 +1838,54 @@ pub fn resolve_path(
     }
 }
 
+/// resolve paths like ::path::to::file
+fn resolve_global_path(
+    path: &RacerPath,
+    filepath: &Path,
+    search_type: SearchType,
+    namespace: Namespace,
+    session: &Session,
+    import_info: &ImportInfo,
+) -> Vec<Match> {
+    if path.segments.len() < 1 {
+        return Vec::new();
+    }
+    let context = search_crate_root(
+        &path.segments[0],
+        filepath,
+        SearchType::ExactMatch,
+        namespace,
+        session,
+        import_info,
+        false,
+    )
+    .into_iter()
+    .nth(0);
+    if let Some(context) = context {
+        if path.segments.len() == 1 {
+            let mut v = Vec::new();
+            v.push(context);
+            return v;
+        }
+        let mut new_path = path.clone();
+        new_path.prefix = None;
+        match context.mtype {
+            MatchType::Module => resolve_path(
```

Comment:
Here `MatchType::Crate` can be available so please use `resolve_following_path` here.